syft/frameworks/hook.py

```diff
@@ -0,0 +1,19 @@
+from abc import ABC
+from abc import abstractmethod
+
+from syft.workers import BaseWorker
+
+
+class BaseHook(ABC):
```

Comment:
Should this file live in abstract rather than frameworks?

Reply:
I wanted to keep it separate, since `abstract.py` is meant for tensor abstraction. I think that file deserves to be renamed!

Reply:
in the most recent series of commits, I split this file into an object.py and a tensor.py, and put both under the syft/generic directory

Reply:
Following up on this, should this file live in generic then instead of frameworks?

Reply:
In my opinion no, because it's part of the framework integration spec outlined in OpenMined/rfcs#3, and I was hoping to keep those kinds of integration points self-contained

Reply:
makes sense to me